src/context/MediaContext.tsx

```diff
@@ -0,0 +1,164 @@
+'use client'
+
+import React, {
+  createContext,
+  useContext,
+  ReactNode,
+  useState,
+  useEffect
+} from 'react'
+
+const MediaContext = createContext<
+  | {
+      medias: MediaProps[]
+      currentMedia: number
+      setCurrentMedia: (value: number) => void
+      isPlaying: boolean
+      setIsPlaying: (isPlaying: boolean) => void
+      handleRemove: (index: number) => void
+      playStatus: PlayMode
+      setPlayStatus: (playStatus: number) => void
+      handleSongEnded: () => void
+    }
+  | undefined
+>(undefined)
+
+interface MediaProps {
+  url: string
+  avatar: string
+  name: string
+  localUrl?: string
+  duration?: number
+}
+
+interface MediaProviderProps {
+  children: ReactNode
+}
+
+const mockData = [
+  {
+    avatar: '/avatar.png',
+    name: 'horse.mp3',
+    url: 'https://ipfs.io/ipfs/QmeVPhqpAhJcdeP6CSFi75GHRFY8HHh7k5Vsn5fAg5SajZ',
+    duration: 0
+  },
+  {
+    avatar: '/avatar.png',
+    name: 'cat.mp3',
+    url: 'https://ipfs.io/ipfs/QmWGAPKwjUw1AqsEYnYjZyePnGXnBhGfB21D4eeZzFwzhZ',
+    duration: 0
+  },
+  {
+    avatar: '/avatar.png',
+    name: 'bird.mp3',
+    url: 'https://ipfs.io/ipfs/QmeVPhqpAhJcdeP6CSFi75GHRFY8HHh7k5Vsn5fAg5SajZ',
+    duration: 0
+  },
+  {
+    avatar: '/avatar.png',
+    name: 'dog.mp3',
+    url: 'https://ipfs.io/ipfs/QmWGAPKwjUw1AqsEYnYjZyePnGXnBhGfB21D4eeZzFwzhZ',
+    duration: 0
+  }
+]
+
+export enum PlayMode {
+  CYCLE = 0,
+  INFINITE = 1,
+  RANDOM = 2
+}
+
+export const MediaProvider = ({ children }: MediaProviderProps) => {
+  const [medias, setMedias] = useState<MediaProps[]>(mockData)
+  const [currentMedia, setCurrentMedia] = useState<number>(0)
+  const [isPlaying, setIsPlaying] = useState<boolean>(false)
+  const [playStatus, setPlayStatus] = useState<number>(0)
+
+  useEffect(() => {
+    const fetchFile = async () => {
```

Comment:
Did you find it necessary to manually fetch the file for playback instead of just setting the audio source? What challenges did you encounter?